src/PgRvn.Server/Messages/MessageReader.cs

```diff
@@ -360,9 +369,23 @@ private async Task<byte> ReadByteAsync(PipeReader reader, CancellationToken toke
             return ReadByte(read.Buffer, reader);
         }
 
+        private async Task SkipBytesAsync(PipeReader reader, CancellationToken token, int length)
+        {
+            var read = await ReadMinimumOf(reader, length, token);
+            SkipBytes(read.Buffer, reader, length);
+        }
+
+        private void SkipBytes(ReadOnlySequence<byte> readBuffer, PipeReader reader, int length)
+        {
+            var sequence = readBuffer.Slice(0, length);
+            reader.AdvanceTo(sequence.End);
+        }
+
         private byte[] ReadBytes(ReadOnlySequence<byte> readBuffer, PipeReader reader, int length)
         {
             var sequence = readBuffer.Slice(0, length);
+
+            var buffer2 = ArrayPool<byte>.Shared.Rent(length);
```

Comment:
huh? Why do you have this and still retain the old buffer?
Also, where do you intend to return it?

Also, note that if you rent an array, it may come up *bigger* than expected.

Reply:
Not sure why that was added, removing it